src/indra_cogex/sources/indra_db/raw_export.py

```diff
@@ -1,369 +1,64 @@
-            if ("pubmed", "abstract") in reading_id_per_type:
-                drop.add(reading_id_per_type[("pubmed", "abstract")])
-            if ("pubmed", "title") in reading_id_per_type:
-                drop.add(reading_id_per_type[("pubmed", "title")])
-    # In case of fulltext, we can drop all non-fulltexts, and then drop
-    # everything that is lower on the fulltext priority order
-    else:
-        priority = [
-            "xdd-pubmed",
-            "xdd",
-            "xdd-biorxiv",
-            "cord19_pdf",
-            "elsevier",
-            "cord19_pmc_xml",
-            "manuscripts",
-            "pmc_oa",
-        ]
-        drop |= {c[0] for c in not_fulltexts}
-        sorted_fulltexts = sorted(
-            fulltexts, key=lambda x: priority.index(x[2]), reverse=True
-        )
-        drop |= {c[0] for c in sorted_fulltexts[1:]}
-    return drop
-
-
-def load_text_refs_by_trid(fname: str):
-    text_refs = {}
-    for line in tqdm.tqdm(
-        gzip.open(fname, "rt", encoding="utf-8"),
-        total=36223169,
-        unit_scale=True,
-        desc="Processing text refs into a lookup dictionary",
-    ):
-        ids = line.strip().split("\t")
-        id_names = ["TRID", "PMID", "PMCID", "DOI", "PII", "URL", "MANUSCRIPT_ID"]
-        d = {}
-        for id_name, id_val in zip(id_names, ids):
-            if id_val != "\\N":
-                if id_name == "TRID":
-                    id_val = int(id_val)
-                d[id_name] = id_val
-        text_refs[int(ids[0])] = d
-    return text_refs
-
-
-def get_update(start_date):
-    # start date e.g., '20220112 00:00:00'
-    query = """
-        SELECT tc.insert_date, tr.pmid, tr.pmcid
-        FROM public.text_content as tc, public.text_ref as tr
-        WHERE tc.text_ref_id = tr.id
-        AND tc.insert_date >= start_date
-    """
-
+def get_latest_timestamp_prefix(bucket: str, prefix: str) -> str:
+    import boto3
+    from datetime import datetime
+    s3 = boto3.client("s3")
+    paginator = s3.get_paginator("list_objects_v2")
+    result = paginator.paginate(Bucket=bucket, Prefix=prefix, Delimiter="/")
+    timestamps = []
+    for page in result:
+        for cp in page.get("CommonPrefixes", []):
+            folder_name = cp["Prefix"].split("/")[-2]
+            try:
+                timestamps.append(datetime.strptime(folder_name, "%Y%m%d-%H%M%S"))
+            except ValueError:
+                continue
+    if not timestamps:
+        raise ValueError("No valid timestamped folders found in S3.")
+
+    latest_ts = max(timestamps).strftime("%Y%m%d-%H%M%S")
+    return f"{prefix}{latest_ts}/"
+
+def download_s3_file(bucket: str, s3_key: str, local_path: Path):
+    import boto3
+    s3 = boto3.client("s3")
+    local_path.parent.mkdir(parents=True, exist_ok=True)
+    s3.download_file(Bucket=bucket, Key=s3_key, Filename=str(local_path))
+    logger.info(f"Downloaded s3://{bucket}/{s3_key} → {local_path}")
```

Comment:
There might be some function that can be reused from https://github.com/sorgerlab/indra/blob/master/indra/util/aws.py